pkgs/os-specific/linux/compsize/default.nix

```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchFromGitHub, btrfs-progs }:
+
+stdenv.mkDerivation rec {
+  name = "compsize-${version}";
+  version = "2018-04-07";
+
+  src = fetchFromGitHub {
+    owner = "kilobyte";
+    repo = "compsize";
+    rev = "903f772e37fc0ac6d6cf94ddbc98c691763c1e62";
+    sha256 = "0jps8n0xsdh4mcww5q29rzysbv50iq6rmihxrf99lzgrw0sw5m7k";
+  };
+
+  nativeBuildInputs = [ btrfs-progs ];
```

Comment:
This should be `buildInputs` (not `nativeBuildInputs`): the actual btrfs tools don't need to run on the machine where you build this but the build includes some btrfs headers for the target platform. Doesn't make a difference unless you're cross-building but let's get this right anyway. 

The distinction is rather subtle, see https://nixos.org/nixpkgs/manual/#ssec-stdenv-dependencies.
Simple rule of thumb: if it needs to _run_ during the build process, put it in `nativeBuildInputs`.